File: 57-exercises-for-programmers/c-sharp/exercise_01.cs
// Create a program that prompts for your name and prints a greeting using your
// name.

public class Exercise1 {
	string Name { get; set; }

	public Exercise1() {
		PromptName();
		PrintName();
	}

	void PromptName() {
		System.Console.Write("What's your name? ");
		Name = System.Console.ReadLine();
	}

	void PrintName() {
		System.Console.WriteLine("Hi, {0}!", Name);
	}
}

File: 57-exercises-for-programmers/rust/src/exercise_01.rs
// Create a program that prompts for your name and prints a greeting using your
// name.

pub mod exercise_1 {
    pub fn run() {
        display_greeting();
    }

    fn display_greeting() {
        use std::io::{stdin, stdout, Write};

        print!("What is your name? ");
        stdout().flush().unwrap();

        let mut name = String::new();
        stdin()
            .read_line(&mut name)
            .expect("Error: Invalid Input.");

        println!("Hello, {}, nice to meet you!", name.trim());
    }
}
